corehq/apps/reports/management/commands/prepare_projects_for_case_list_explorer.py

```diff
@@ -0,0 +1,111 @@
+            dest='domain',
+            help='Specify a specific project to run data a integrity check and re-sync on.'
+        )
+
+    def handle(self, **options):
+        domain = options.pop('domain')
+        if domain is not None:
+            if DomainsNotInCaseSearchIndex.objects.count() > 0:
+                self.stdout.write("Please finish running the complete initial sync by "
+                                  "running this management command without the --project "
+                                  "argument.")
+                return
+            self.run_domain_re_sync(domain)
+            return
+
+        self.migrate_domains_not_in_case_search_index()
+        self.stdout.write("Initial sync complete.")
+        self.run_data_integrity_checks()
+
+    def migrate_domains_not_in_case_search_index(self):
+        count = DomainsNotInCaseSearchIndex.objects.count()
+        if count > 0:
+            self.stdout.write(f"\n\nFound {count} project(s) to sync...")
+
+            confirm = input("Please make sure you run this command inside a terminal multiplexer "
+                            "like tmux or screen as this could take some time. Ready to continue? [y/n] ")
+            if confirm.lower() != 'y':
+                self.stdout.write("Aborting initial sync...")
+                return
+        else:
+            return
+
+        # delete DomainsNotInCaseSearchIndex for zero-case domains
+        DomainsNotInCaseSearchIndex.objects.filter(estimated_size__lte=0).all().delete()
+
+        for tracker in DomainsNotInCaseSearchIndex.objects.order_by('estimated_size').all():
+            self.stdout.write(f"\n\nSyncing {tracker.domain}'s with {tracker.estimated_size} cases...")
+            domain_needs_search_index.clear(tracker.domain)
+            DomainsNotInCaseSearchIndex.objects.filter(domain=tracker.domain).delete()
+            CaseSearchReindexerFactory(domain=tracker.domain).build().reindex()
+            self.stdout.write(f"\nCompleted syncing {tracker.domain}...\n-----\n")
+
+    def run_domain_re_sync(self, domain, total_case=None, total_case_search=None):
+        if total_case is None:
+            total_case = CaseES().domain(domain).count()
+            total_case_search = CaseSearchES().domain(domain).count()
+            self.stdout.write(f"\nProject {domain} has {total_case} cases in CaseIndex "
+                              f"and {total_case_search} cases in CaseSearchIndex.")
+            confirm = input("Would you like to re-sync this project? [y/n] ")
+            if confirm.lower() != 'y':
+                self.stdout.write("Aborting data integrity sync...")
+                return
+
+        self.stdout.write(f"\n\nRe-syncing {domain} with {total_case} cases...")
+        CaseSearchReindexerFactory(domain=domain).build().reindex()
+        self.stdout.write(f"\nCompleted syncing {domain}.\n"
+                          f"Previous total synced was {total_case_search}...\n-----\n")
```

Comment:
Functions should be as agnostic of their caller's environment as possible. The way this signature is written, `total_case` and `total_case_search` should be independent of each other, i.e., I should be able to supply `total_case`, `total_case_search`, or neither. The logic here, however, suggests that `total_case_search` is expected to be supplied if `total_case` is. I'd separate out their population so they are done individually. I'd also consider whether the population and prompting logic belongs in this function, as `run_data_integrity_checks` does this logic on its own, whereas `handle` depends on the logic here. It might be more consistent if it was one or the other.